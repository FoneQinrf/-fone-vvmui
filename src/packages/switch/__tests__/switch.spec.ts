/*
 * @Author: Fone`峰
 * @Date: 2021-05-17 17:37:01
 * @LastEditors: Fone`峰
 * @LastEditTime: 2021-05-20 16:08:45
 * @Description: file content
 * @Email: qinrifeng@163.com
 * @Github: https://github.com/FoneQinrf
 */
import { mount } from '@vue/test-utils';
import Switch from '../index.vue';
import { nextTick } from 'vue';

describe('Switch', () => {
  const wrapper = mount(Switch);
  it('should render type corretly', async () => {
    wrapper.setProps({ type: 'default' });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('--default');

    wrapper.setProps({ type: 'error' });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('--error');

    wrapper.setProps({ type: 'success' });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('--success');

    wrapper.setProps({ type: 'warning' });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('--warning');
  });

  it('should render active and click corretly', async () => {
    expect(wrapper.find('.vvm-switch--radius.--close').exists()).toBe(true);
    wrapper.trigger('click');
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.classes()).toContain('--active');
    expect(wrapper.emitted()).toHaveProperty('change');
    expect(wrapper.find('.vvm-switch--radius.--active').exists()).toBe(true);
  });

  it('should render disabled coreetly', async () => {
    expect(wrapper.find('.vvm-switch--disabled').exists()).toBe(false);
    wrapper.setProps({ disabled: true });
    await nextTick();
    expect(wrapper.html()).toMatchSnapshot();
    expect(wrapper.find('.vvm-switch--disabled').exists()).toBe(true);
    wrapper.trigger('click');
    await nextTick();
    expect(wrapper.vm.currentValue).toBe(true);
  });
});
