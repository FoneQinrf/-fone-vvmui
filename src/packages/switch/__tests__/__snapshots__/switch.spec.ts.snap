// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch should render active and click corretly 1`] = `
"<label class=\\"vvm-switch --warning --active\\"><span class=\\"vvm-switch--radius --active\\"></span>
  <!--v-if-->
</label>"
`;

exports[`Switch should render disabled coreetly 1`] = `"<label class=\\"vvm-switch --warning --active\\"><span class=\\"vvm-switch--radius --active\\"></span><span class=\\"vvm-switch--disabled\\"></span></label>"`;

exports[`Switch should render type corretly 1`] = `
"<label class=\\"vvm-switch --default\\"><span class=\\"vvm-switch--radius --close\\"></span>
  <!--v-if-->
</label>"
`;

exports[`Switch should render type corretly 2`] = `
"<label class=\\"vvm-switch --error\\"><span class=\\"vvm-switch--radius --close\\"></span>
  <!--v-if-->
</label>"
`;

exports[`Switch should render type corretly 3`] = `
"<label class=\\"vvm-switch --success\\"><span class=\\"vvm-switch--radius --close\\"></span>
  <!--v-if-->
</label>"
`;

exports[`Switch should render type corretly 4`] = `
"<label class=\\"vvm-switch --warning\\"><span class=\\"vvm-switch--radius --close\\"></span>
  <!--v-if-->
</label>"
`;
